libs/plugin-scanner/src/main/java/org/elasticsearch/plugin/scanner/ClassReadersProvider.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugin.scanner;
+
+import org.objectweb.asm.ClassReader;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A utility class containing methods to create streams of ASM's ClassReader
+ *
+ * @see ClassReader
+ */
+public class ClassReadersProvider {
+    private static final String MODULE_INFO = "module-info.class";
+
+    public List<ClassReader> ofClassPath() throws IOException {
```

Comment:
What in this class requires some of these methods to be instance methods? I don't see a ctor or any instance member vars. I suspect this was for mocking; see my other comment. Note that it is especially odd to make some methods here instance methods, but still also have static utility methods.

Reply:
yes, this was only made an instance to allow mocking this